### PR TITLE
fix: dispatch OnAfterMessageSentEvent after streaming delivery

### PR DESCRIPTION
**Pull request by bingchengcc**
## Problem

When streaming output is active, `OnAfterMessageSentEvent` was never dispatched because the streaming path returned early before reaching the event hook at the bottom of `process()`.

## Solution

- Extract `_after_sent_cleanup()` helper that dispatches `OnAfterMessageSentEvent` and clears the result
- Call it in both the streaming path and the regular path

Rebased onto latest master.

## Test plan

- [x] Unit tests pass
- [x] Manual test: streaming message now triggers `OnAfterMessageSentEvent`

## Summary by Sourcery

Ensure streaming message delivery completes the same post-send lifecycle as regular responses.

Bug Fixes:
- Dispatch `OnAfterMessageSentEvent` consistently after successfully delivered streaming messages, matching regular message delivery behavior.

Enhancements:
- Standardize streaming delivery results across supported platform event implementations so the pipeline can distinguish successful delivery from empty or unsuccessful streams.
- Centralize post-delivery event handling and result cleanup for both streaming and regular responses.

### CODE REVIEW

File: astrbot/core/pipeline/respond/stage.py [hunk 1]
```diff
@@ -228,6 +234,8 @@ async def process(
             )
             logger.info(f"Applying streaming output ({event.get_platform_id()}).")
             await event.send_streaming(result.async_stream, realtime_segmenting)
+            if await self._after_sent_cleanup(event):
```
[the remark this answers is not in the document]
Comment (bingchengcc) [reply astrbot/core/pipeline/respond/stage.py]: Thanks for the detailed review. For the current codebase, all streaming adapters (Telegram, aiocqhttp, Discord, etc.) fully `await` the streaming delivery before `send_streaming` returns. For example, Telegram does `await self._send_streaming_draft(...)` / `await self._send_streaming_edit(...)` — the `asyncio.create_task(Metric.upload(...))` that follows is only for metrics reporting, not message delivery. So the event is dispatched after all chunks are actually sent. That said, this is a valid concern for future adapters that might use a fire-and-forget pattern, and worth documenting in the adapter interface contract.

[the remark this answers is not in the document]
Comment (bingchengcc) [reply astrbot/core/pipeline/respond/stage.py]: Good catch. In practice this only happens when the model produces a completely empty stream (no chunks at all), which is rare. The `async_stream is None` case is already handled above. A clean fix would be to have `send_streaming` return a bool indicating whether anything was delivered, but that changes the adapter interface for all platforms — probably better as a follow-up.

**Comment by bingchengcc** [comment]
Hi! Could you please approve the workflow runs? I'm an external contributor so the CI is stuck at `action_required`. Rebased onto latest master, single commit, 9+/3-.

@buyun14 @Soulter

**Review by kilisamemarisaaa — CHANGES_REQUESTED** [review]
### Empty streams are reported as sent

At head `2f91e54d8a90a5a81156c3d1ef6c4684ad19d080`, the streaming branch now always calls `_after_sent_cleanup(event)` after `await event.send_streaming(...)`. However, the existing `aiocqhttp` implementation returns early when its async generator yields no chains, before it sends anything. The new hook then dispatches `OnAfterMessageSentEvent` and clears `event.result` even though no message was delivered.

That changes the event contract for empty/filtered responses: plugins can perform post-send work (logging, persistence, follow-up actions) for a message that never existed, and the result is lost. Please make the streaming API return an explicit delivery outcome (or otherwise retain the existing `has_sent` signal) and invoke cleanup only when at least one message was actually delivered. A regression test should exercise an empty async stream through the real stage path and assert that the hook is not called and the result remains available.

This is separate from the timing concern in the automated review: it applies even when the adapter awaits all delivery work. I did not claim a full adapter matrix or run the project suite; the finding is based on the current `aiocqhttp.send_streaming` early return and the changed stage control flow.

**Comment by bingchengcc** [comment]
Thanks for the detailed review! Fixed in the new commit.

**What changed:**

- `send_streaming` now returns `bool` instead of `None`. The base class (`AstrMessageEvent`) returns `True` (it marks `_has_send_oper` and assumes delivery). All buffering adapters that early-return on empty generators (`if not buffer: return None`) now return `False` instead.
- `RespondStage.process()` captures the return value and only invokes `_after_sent_cleanup(event)` when `delivered=True`:

```python
delivered = await event.send_streaming(result.async_stream, realtime_segmenting)
if delivered:
    if await self._after_sent_cleanup(event):
        return
    return
```

So an empty/filtered stream (e.g. aiocqhttp's generator yielding no chains) no longer dispatches `OnAfterMessageSentEvent` or clears `event.result`.

**Affected adapters:** aiocqhttp, dingtalk, discord, line, misskey, slack, mattermost, wecom, weixin_official_account, weixin_oc (buffering pattern), plus lark/telegram/qqofficial/webchat/wecom_ai_bot (complex streaming, explicit `return True` on delivery, `return False` on early empty exit).

Happy to add a dedicated regression test for the empty-stream path if you'd like, though the existing pipeline tests cover the stage flow.

**Review by kilisamemarisaaa — APPROVED** [review]
Reviewed updated head 450b02c4b35eb4759caed3c3611c893c48762f4b. The stage now gates _after_sent_cleanup on the bool returned by send_streaming, and buffering adapters return False on empty streams while successful delivery paths return True or delegate to the base implementation. This resolves the empty-stream lifecycle issue I reported. I did not claim a full local test run because the repository dependencies are unavailable in this environment.
